README
MPDCon originally written by : Daniel Luederwald
Forked from : http://ftp.debian.org/debian/pool/main/m/mpdcon.app/mpdcon.app_1.1.99.orig.tar.gz
Just some little changes to avoid chrashes using GNUstep trunk and clang



ChangeLog
2004-06-14  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* OwnTableView.m, FindController.m: Code Reformatting

	* CrossfadeController.m, CollectionController.m, AppController.m:
	Reformatted Code

2004-06-13  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* FindController.m, CollectionController.m, CollectionController.h:
	*** empty log message ***

	* AppController.m, AppController.h: Implemented getAlbums service

	* FindController.m, FindController.h, CrossfadeController.m, CrossfadeController.h, CollectionController.m, CollectionController.h, MPDController.m, MPDController.h:
	Refactoring

	* PlaylistController.m, PlaylistController.h: More Refactoring

	* PlayView.m, PlayView.h: Refactoring

	* AppController.h: *** empty log message ***

	* AppController.h, AppController.m: More Refactoring

2004-06-12  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* OwnPanel.h, OwnPanel.m: *** empty log message ***

	* OwnPanel.h, OwnPanel.m: New file.

2004-06-04  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* FindController.m, FindController.h, AppController.m: Bugfix

	* AppController.m: *** empty log message ***

	* PlaylistsManagerController.m: Begin of localization

	* PlaylistController.m: *** empty log message ***

	* PlayView.m, MPDController.m, FindController.m, AppController.m:
	Begin of localization

	* Strings.h: New file.

	* StatisticsController.h, PreferencesController.h, PlaylistsManagerController.h, PlaylistController.h, PlayView.h, OwnTableView.h, MPDController.h, FindController.h, CrossfadeController.h, CollectionController.h:
	*** empty log message ***

	* AppController.h: Service Method

	* StatisticsController.m, PreferencesController.m, PlaylistsManagerController.m, PlaylistController.m, PlayView.m, MPDController.m, FindController.m:
	*** empty log message ***

	* CollectionController.m: Find Capability

	* AppController.m: Service Method

2004-06-02  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* AppController.m, FindController.m: *** empty log message ***

	* FindController.h, AppController.h, FindController.m, AppController.m:
	Refactoring

	* CollectionController.m, FindController.m, FindController.h, AppController.m, CollectionController.h, PlaylistController.h:
	*** empty log message ***

	* CollectionController.h, PlaylistController.m: Find Capability

	* CollectionController.m: Find Capabilities

	* FindController.h, FindController.m: *** empty log message ***

2004-06-01  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* CollectionController.m: *** empty log message ***

	* CollectionController.m, PlaylistController.m, CollectionController.h:
	Implemented Find Capabilities

	* PlaylistController.h: *** empty log message ***

	* PlaylistController.h: Implemented Find Capabilities

	* FindController.h: *** empty log message ***

	* AppController.h, AppController.m: Added Find Capabilities

	* FindController.m, FindController.h: New file.

2004-05-30  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* PlaylistController.m: Highlight moved Song

	* OwnTableView.h: Bugfix

	* OwnTableView.h, OwnTableView.m: New file.

2004-05-27  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* PlaylistController.m: *** empty log message ***

	* PreferencesController.h: Minor Bugfix

	* MPDController.m: More Refactoring

2004-05-26  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* MPDController.m: Bugfix

	* MPDController.h, MPDController.m, PlaylistController.m, PlaylistController.h, PlayView.m, PlayView.h, AppController.m:
	Refactoring

	* AppController.h, PreferencesController.h: *** empty log message ***

	* PreferencesController.m: Refactoring

	* PlaylistItem.h: *** empty log message ***

	* PlaylistItem.m: Refactoring

	* CollectionController.h: *** empty log message ***

	* CollectionController.m, PlaylistsManagerController.h, PlaylistsManagerController.m:
	Refactoring

	* StatisticsController.h: *** empty log message ***

	* StatisticsController.m: Refactoring

	* StatisticsItem.h: *** empty log message ***

	* StatisticsItem.m: Refactoring

	* CrossfadeController.h: *** empty log message ***

	* CrossfadeController.m, AppController.m: Refactoring

2004-05-22  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* MPDController.h, MPDController.m:
	Refactoring + Handling of missing Tags

	* CrossfadeController.m: *** empty log message ***

	* PreferencesController.m: Bugfix

	* AppController.m: *** empty log message ***

2004-05-21  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* AppController.m, AppController.h: *** empty log message ***

	* AppController.m, AppController.h:
	Remember PlayerWindow screen-location

	* PlayView.h, PlayView.m: More Refactoring + Bugfix

2004-05-20  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* PlayView.m: Modified Scrolling

	* AppController.m, AppController.h, CrossfadeController.h, StatisticsItem.h:
	Refactoring

	* StatisticsController.h: *** empty log message ***

	* StatisticsController.m, StatisticsController.h: Refactoring

	* PlaylistsManagerController.m, PlaylistsManagerController.h, CollectionController.m, CollectionController.h:
	*** empty log message ***

	* PlaylistsManagerController.m: Refactoring

	* PlaylistsManagerController.h: *** empty log message ***

	* PlaylistsManagerController.h: Refactoring

2004-05-19  Daniel Luederwald  <flip@Confusion.CabalOfEntropy.local>

	* CollectionController.m, CollectionController.h, PlaylistItem.h, PlayView.h, PlayView.m:
	Refactoring

	* PreferencesController.m: Refactoring + Bugfix

	* PreferencesController.h: Bugfix

	* PreferencesController.h, PlaylistController.m, PlaylistController.h, MPDController.h, MPDController.m:
	Refactoring

	* CrossfadeController.h: New file.

	* MPDController.h, AppController.h, AppController.m, MPDController.m:
	Implemented Crossfade

	* CrossfadeController.m: New file.


AppController.h
/* 
   Project: MPDCon

   Copyright (C) 2004

   Author: Daniel Luederwald

   Created: 2004-05-12 17:59:14 +0200 by flip
   
   Application Controller

   This application is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public
   License as published by the Free Software Foundation; either
   version 2 of the License, or (at your option) any later version.
 
   This application is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   Library General Public License for more details.
 
   You should have received a copy of the GNU General Public
   License along with this library; if not, write to the Free
   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111 USA.
*/
 
#ifndef _PCAPPPROJ_APPCONTROLLER_H
#define _PCAPPPROJ_APPCONTROLLER_H

#include <AppKit/AppKit.h>
#include "PlayView.h"
#include "PreferencesController.h"
#include "PlaylistController.h"
#include "PlaylistItem.h"
#include "MPDController.h"
#include "PlaylistsManagerController.h"
#include "StatisticsController.h"
#include "CrossfadeController.h"
#include "Strings.h"

#define player_MPD 0
#define player_CD 1

@interface AppController : NSObject
{
  IBOutlet NSButton *playButton;
  IBOutlet NSButton *stopButton;
  IBOutlet NSButton *prevButton;
  IBOutlet NSButton *nextButton;
  IBOutlet NSButton *shuffleButton;
  IBOutlet NSButton *repeatButton;
  IBOutlet PlayView *playView;
  IBOutlet NSSlider *percentSlider;
  IBOutlet NSSlider *volumeSlider;

  IBOutlet NSWindow *window;

  IBOutlet NSMenuItem *statisticsItem;
  IBOutlet NSMenuItem *browseItem;
  IBOutlet NSMenuItem *updateItem;
  IBOutlet NSMenuItem *showItem;
  IBOutlet NSMenuItem *manageItem;
  

  NSTimer *anTimer;

  int player;
  int playedSong;
  int prevState;

  BOOL connected;
  BOOL didDisconnect;

  MPDController *mpdController;
}

// Gui Methods
- (void) showPrefPanel:(id)sender;
- (void) showPlaylist: (id)sender;
- (void) managePlaylists: (id)sender;
- (void) browseCollection: (id)sender;
- (void) serverStatistics: (id)sender;
- (void) showCrossfade: (id)sender;

- (void) connect: (id)sender;

- (void) updateCollection: (id)sender;

- (void) play: (id)sender;
- (void) stop: (id)sender;
- (void) prev: (id)sender;
- (void) next: (id)sender;
- (void) shuffle: (id)sender;
- (void) repeat: (id)sender;
- (void) percentChanged: (id)sender;
- (void) volumeChanged: (id)sender;

// Notification Methods

- (void) didDisconnect: (NSNotification *)notif;
- (void) didConnect: (NSNotification *)notif;
- (void) prefsChanged: (NSNotification *)notif;

// Service Methods
- (void) getPlaylist: (NSPasteboard*)pboard 
            userData: (NSString*)userData 
               error: (NSString**)error;

- (void) getAlbums: (NSPasteboard*)pboard 
          userData: (NSString*)userData 
             error: (NSString**)error;

@end

#endif
